utils/src/main/scala/za/co/absa/enceladus/utils/testUtils/HadoopFsTestBase.scala

```diff
@@ -13,12 +13,22 @@
  * limitations under the License.
  */
 
-package za.co.absa.enceladus
 
-import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+package za.co.absa.enceladus.utils.testUtils
 
-trait S3DefaultCredentialsProvider {
+import org.apache.hadoop.fs.FileSystem
+import org.apache.spark.sql.SparkSession
+import za.co.absa.enceladus.utils.fs.FileSystemUtils.FileSystemExt
 
-  implicit val defaultCredentialsProvider = DefaultCredentialsProvider.create()
 
+trait HadoopFsTestBase extends HasSparkSession {
```

Comment:
Could this make it easier for us?

```suggestion
trait HadoopFsTestBase extends SparkTestBase {
```

Reply:
Well, even though I don't think this is the best approach, I tried to do it. Thinking, well, ok, would be fine for tests, I guess. Fewer traits, less mess. But then you stumble upon cases like `za.co.absa.enceladus.examples.CustomRuleSample1` and you realize that some tests might have justified reasons why to provide their own spark session while still needing to test the HadoopFs features, too. In those cases, it would not be correct to base HadoopFsTestBase on SparkTestBase directly, because that already brings its own spark session that might differ from what's needed there. 

I know you could override the spark-session in those cases, too, but that seems quite against the logic of the unit tests, they should IMHO be created on the simples grounds possible and to have anything unrelated dragged in. :(